posthog/models/feature_flag/flag_matching.py

```diff
@@ -489,22 +493,24 @@ def get_all_feature_flags(
         # with a small timeout
         try:
             with execute_with_timeout(FLAG_MATCHING_QUERY_TIMEOUT_MS):
-                if person_id is None:
-                    # :TRICKY: Some ingestion delays may mean that `$identify` hasn't yet created
-                    # the new person on which decide was called.
-                    # In this case, we can try finding the person_id for the old distinct id.
-                    # This is safe, since once `$identify` is processed, it would only add the distinct_id to this
-                    # existing person. If, because of race conditions, a person merge is called for later,
-                    # then https://github.com/PostHog/posthog/blob/master/plugin-server/src/worker/ingestion/person-state.ts#L421
-                    # will take care of it^.
-                    person_id = (
-                        PersonDistinctId.objects.filter(distinct_id=hash_key_override, team_id=team_id)
-                        .values_list("person_id", flat=True)
-                        .first()
-                    )
-                    # If even this old person doesn't exist yet, we're facing severe ingestion delays
-                    # and there's not much we can do, since all person properties based feature flags
-                    # would fail server side anyway.
+                # :TRICKY: Some ingestion delays may mean that `$identify` hasn't yet created
+                # the new person on which decide was called.
+                # In this case, we can try finding the person_id for the old distinct id.
+                # This is safe, since once `$identify` is processed, it would only add the distinct_id to this
+                # existing person. If, because of race conditions, a person merge is called for later,
+                # then https://github.com/PostHog/posthog/blob/master/plugin-server/src/worker/ingestion/person-state.ts#L421
+                # will take care of it^.
+
+                # Also, we disregard the person_id from L467 above because it's possible that this person_id was deleted
+                # before we get here.
```

Comment:
```suggestion
                # Also, we disregard the person_id from L467 above and query again because it's possible that this person_id was deleted
                # before we get here.
                # Further, with the new merging logic, the merge can happen into either of the original persons.
```